### PR TITLE
Add donate_url and opencollective options to add donate buttons to the sidebar

### PR DESCRIPTION
**Pull request by sloria**
When either are set, adds a "Donate" section to the sidebar.
If `donate_url` is set, a "donate" shield button is used.
If `opencollective` is set, an OpenCollective button is used.

With `donate_url = ""https://www.paypal.me/StevenLoria"`:

donate button

With `opencollective = "marshmallow"`:

opencollective button

Both may be set:

both buttons

I also removed the underlines from badges in the sidebar, which I assumed were inadvertent:

**Before**

with underlines

**After**

without underlines

### CODE REVIEW

**Comment by sloria** [comment]
Some notes:

* The Gratipay options should be deprecated or removed, now that Gratipay has shut down
* This will conflict with https://github.com/bitprophet/alabaster/pull/142 . I'll take care of rebasing after that one's merged.
* I didn't update the changelog. Let me know if you'd prefer I do that or if you'd rather do it yourself.
* Not that it concerns you, but if this and #142, I think the [marshmallow docs](https://marshmallow.readthedocs.io) will be able to get off of its [fork of alabaster](https://github.com/sloria/alabaster/tree/sloria)!

**Comment by bitprophet** [comment]
Links: #132, #128 (presumably).

**Comment by sloria** [comment]
Ah, yes. Sorry, I should have looked at those existing issues/PRs. I believe this should close #132 (but not #128). Support for other donation button types (e.g. for Patreon, Stripe, etc.) can always be added in the future.

**Comment by bitprophet** [comment]
OK, INI syntax doesn't seem to do lists. We could make it a `conf.py` level setting, but I think at this point I don't care enough. If donation.html gets too heavy we can always opt to generify it later, perhaps. Though that's harder.

**Comment by bitprophet** [comment]
Grumble, one issue with the "handful of orthogonal donation buttons" approach is it'll make the hide/show `{% if %}` statement get completely ridiculous. Guess we could at least move that logic to the top of `alabaster.css_t` or something like we do with some other things, but still.

Actually...if we avoid the (admittedly, would be nice) flavor text and header at top/bottom - in other words, making `donation.html` just a loose collection of independently shown/hidden elements - that solves the problem neatly. It's also just kinda how the sidebar works overall right now.

I don't _like_ that fact, but I'm willing to punt to a general overhaul of the sidebar sometime later.

**Comment by bitprophet** [comment]
Hrm, the screenshot from the OP doesn't match what I am seeing re: multiple donate buttons in play, there's zero padding between them in my case:

image

I'm pulling in Steven's code piecemeal but I double checked and I don't seem to be missing any styling or elements 🤔

**Comment by bitprophet** [comment]
OK this should be done, with the abovementioned customizations, will release soonish.

**Comment by bitprophet** [comment]
Nope, that's not it (check off "don't show ads" under Advertising in RTD admin), but partly because the element is still there, just empty. But I am seeing the `::before` pseudo-element in the source view, so...hrm.

Same issue in Safari so it's not just Chrome.

Gonna have to punt on this for now tho because I've already spent so much time here today & it doesn't actually look that "off" unless you knew how it looked on my localhost. Nor is it any worse than the rest of the messed up spacing in the sidebar as a whole (see above re: wanting to overhaul the whole thing).

Still tho. Computers. 😡

File: alabaster/static/alabaster.css_t [hunk 1]
```diff
@@ -221,6 +221,14 @@ div.sphinxsidebar hr {
...
 }
 
+div.sphinxsidebar .badge {
+    border-bottom: none;
+}
+
+div.sphinxsidebar .badge:hover {
+    border-bottom: none;
+}
+
```
Comment (jab) [inline alabaster/static/alabaster.css_t]: I think this whole commit is just working around the root cause, which I fixed in #125 almost a year ago but which still needs to be merged.

(I think the root cause is that some CSS  in the theme that intends to clear the bottom border for all images is accidentally getting overridden by some CSS that comes after it.)

#125 has gotten 👍, but still has not been merged. Once it is merged, I think this workaround can be removed.

/cc @bitprophet @sloria

Comment (jab) [reply alabaster/static/alabaster.css_t]: Hi @bitprophet, my previous comment is still the case, so I just thought I'd see if you might have a moment to merge my PR. Thanks for taking a look if you can!

Comment (jab) [reply alabaster/static/alabaster.css_t]: (#125)

**Pull request closed without merging**